lib/mix/lib/mix/tasks/test.ex

```diff
@@ -642,6 +644,27 @@ defmodule Mix.Tasks.Test do
     Enum.filter(matched_test_files, &MapSet.member?(allowed_files, Path.expand(&1)))
   end
 
+  defp filter_by_partition(files, opts) do
+    if total = opts[:partitions] do
+      partition = System.get_env("MIX_TEST_PARTITION")
+
+      with false <- is_nil(partition),
+           {partition, ""} <- Integer.parse(partition),
+           true <- partition in 1..total do
+        partition = partition - 1
+        Enum.filter(files, &(:erlang.phash2(&1, total) == partition))
```

Comment:
Is there are reason to use hash based partitioning instead of something like:

```elixir
Enum.filter(Enum.with_index(files), fn {_file, index} -> rem(index, total) == partition end)
```

Reply:
The rationale behind hash based partitioning (vs linear & alphabetical ordering) was to get all my slow feature tests (which are located in the same feature folder) randomly distributed over all test runners.

Very excited to see this feature integrated in Elixir, thanks guys!

Reply:
@cblavier in this case, sort+round-robin is actually fairer than hashing, because we will guarantee that, if the feature folder has four files, each file goes to a different partition (assuming at least four partitions). With hashing, they would be distributed unfairly.